### ISSUE TITLE
No error returned when credentials are incorrect

### ISSUE THREAD

**Issue by sraillard** [kubernetes-client/csharp#491]
When calling a method with basic authentication (login and password), if the credentials are incorrect, there is no exception. An example with the following will just produce a `ns` variable with a `null` value:

```
KubernetesClientConfiguration config = new KubernetesClientConfiguration
{
    Username = "xxx",
    Password = "yyy",
    Host = "https://zzz:6443",
    SkipTlsVerify = true
};
Kubernetes client = new Kubernetes(config);
V1Namespace ns = client.ReadNamespace("test");
```

**Comment by sraillard** [comment]
For information, basic auth has been removed in kubernetes 1.19

**Comment by brendandburns** [comment]
I believe that this is intentional. Kubernetes returns 404 if you try to access a namespace that you are unauthorized to access (to prevent namespace probing attacks) so the client interprets that as null.

If you disable anonymous authentication in your Kubernetes cluster, I believe you should get a 403 and then an exception.

If the server is returning 403 and you're not seeing an exception, that is a bug.

**Comment by sraillard** [comment]
How I can get the HTTP status code returned? When a 404 is returned, an exception is thrown. How to check for the 403? I don't have any exceptions thrown and I don't find a method or properties that could tell me the result of the last request.

When the password is incorrect, all these methods are return a `null` object without throwing any exception: `ReadNamespace`, `ReadNamespacedSecret`, `ReadNamespacedServiceAccount`.

**Comment by simonvane** [comment]
@sraillard That's something that took me a while to work out. I've got an example of handling that here - https://github.com/kubernetes-client/csharp/issues/489#issue-714988634

Basically, I catch `HttpOperationException` and look at the response status code within that.

**Comment by sraillard** [comment]
@simonvane : I don't get **any** exception! That is my issue...

**Comment by brendandburns** [comment]
Actually, I looked at the code. It returns null on unauthenticated HTTP 401. It should throw on unauthorized 403 (and on 404, I was wrong above)

This doesn't seem right to me, but it will need to be fixed in the generated code.

**Comment by brendandburns** [comment]
I dug into this some more. This is occurring because the Kubernetes swagger has the following declaration in it:

```json
          "401": {
            "description": "Unauthorized"
          }
```

Which basically means that 401 is a "expected" error and thus it doesn't throw.

I'm going to see if I can strip this from the swagger because it really doesn't make sense to me for 401 not to throw.

**Comment by sraillard** [comment]
You mean it's in the original JSON provided by the Kubernetes project?

**Comment by brendandburns** [comment]
Yes, the swagger.json that is provided by the Kubernetes project, that is then used to generate this library.

There's a pre-processing script that we can use to strip it, I'm making the PR.

**Comment by brendandburns** [comment]
See https://github.com/kubernetes-client/gen/pull/170

**Comment by simonvane** [comment]
> @simonvane : I don't get **any** exception! That is my issue...

@sraillard Sorry, I was hoping using the `...WithHttpMessagesAsync` might throw the exception for you.

**Comment by brendandburns** [comment]
This was fixed by #505
